### ISSUE TITLE
Modal dialog accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7297]
### Description

![Screenshot of an entry selection modal.](https://user-images.githubusercontent.com/17884198/102676946-c8fcc880-4154-11eb-9d97-2057131e0974.png)

Modals are used for element fields (assets, entries, etc.) and the email sent notification on the login page. There may be more instances to identify.

Some ares for accessibility improvements on modals:
- When the modal is open, it should not be possible for users to focus content outside of the modal
- When the modal opens, focus should move to the modal or its first focusable child element
- When the modal closes, focus should return to the element that triggered the modal to open

**References:**
- [W3C aria practices example: Modal Dialog Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html)

### Steps to reproduce
1. Open a modal, e.g. open up an image upload field

### Additional info
- Craft version: 3.5.17.1

**Comment by ajanec01** [comment]
Also, the modal container that customizes the table view in the assets page would benefit from `role="dialog"` and a name. Something in the lines of
```
<form role="dialog" aria-label="customise sources view" ></form>
```

**Comment by gcamacho079** [comment]
We are working to identify other instances of this modal component; accessibility of element select modals (assets, categories, etc.) will be resolved in #10234
